docs/DeveloperGuide.adoc

```diff
@@ -423,6 +423,49 @@ the total cost for that day
 . This results in a new window showing a bar chart of expenses over the last 7 days.
 
 
+// tag::budget[]
+
+=== Budget
+==== Current Implementation
+The current implementation for budget tracking is described as below.
+
+===== Keeping spendings within budget
+The budget mechanism is facilitated by `VersionedAddressBook` and `AddCommand`. Whenever a user adds an expense via
+`AddCommand`m if they do not have a budget set, the application will prompt them to do so. After they do so,
+application proceeds normally. If adding another expense will result in them exceeding the budget, they can still do
+so, but a warning will be given to the user.
+
+===== Recurring budget
+Every time a user log in and `Model#loadUserData()`, the application will check if the user has set a time where the
```

Comment:
Rephrase to

Every time a user logs in, `Model#loadUserData()` is called and the application......